src/index.js

```diff
@@ -1,24 +1,205 @@
 import './style.css';
 
-const mainContainer = document.querySelector('.container');
-const todoTasks = [{
-  description: 'wash the dishes',
-  completed: true,
-  index: 1,
-},
-{
-  description: 'complete to do list project',
-  completed: true,
-  index: 2,
-},
-];
-
-document.addEventListener('DOMContentLoaded', () => {
-  todoTasks.forEach((n) => {
-    const output = document.createElement('div');
-    output.classList.add('listitem');
-    output.setAttribute('id', `${n.index}`);
-    output.innerHTML = `<div><input type="checkbox"> ${n.description}</div> <span class="listitemIcon"><i class="fa-solid fa-ellipsis-vertical"></i></span>`;
-    mainContainer.appendChild(output);
+class Task {
+  constructor(description, index) {
+    this.description = description;
+    this.completed = false;
+    this.index = index;
+  }
+
+  static task() {
+    let tasks;
+    if (localStorage.getItem('tasks') === null) {
+      tasks = [];
+    } else {
+      tasks = JSON.parse(localStorage.getItem('tasks'));
+    }
+    return tasks;
+  }
+
+  static addTask(task) {
+    const taskItem = document.querySelector('textarea');
+    const tasks = this.task();
+    tasks.push(task);
+    taskItem.value = '';
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+    window.location.reload();
+  }
+
+  static showTasks(task) {
+    const taskList = document.getElementById('task-list');
+
+    const div = document.createElement('div');
+    div.setAttribute('draggable', 'true');
+    div.setAttribute('class', 'listed-task');
+    div.setAttribute('id', `${task.index}`);
+    div.setAttribute('completed', `${task.completed}`);
+
+    const check = document.createElement('input');
+    check.setAttribute('type', 'checkbox');
+    check.setAttribute('class', 'check');
+
+    const input = document.createElement('input');
+    input.setAttribute('class', 'input');
+
+    const i = document.createElement('i');
+    i.setAttribute('class', 'bi bi-three-dots-vertical');
+
+    input.value = `${task.description}`;
+    if (task.completed === true) {
+      input.style.textDecoration = 'line-through';
+      check.checked = true;
+    } else {
+      input.style.textDecoration = 'none';
+      check.checked = false;
+    }
+    div.append(check, input, i);
+    taskList.appendChild(div);
+  }
+}
+
+const clear = (event) => {
+  event.preventDefault();
+  let tasks = Task.task();
+  tasks = tasks.filter((t) => t.completed !== true);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  window.location.reload();
+};
```

Comment:
- [ ] When a user deletes a task through the 'clear all completed' button the index of the list is not updated, and if the user creates a task after that, it is possible to have an unordered list of indexes and tasks with the same index. This may leave a bug in your project and it can make the delete function index update not work too. Since it is a project requirement to have the index of the list updated after deletion, I suggest you kindly implement this 'update' functionality to the clear function too.
 
<img width="1015" alt="image" src="https://user-images.githubusercontent.com/61919617/193067432-211c84e7-88c8-4d09-a327-30d42e0ab808.png">

Reply:
@leonardodiasb  i have a problem and i cant solve it , can you plz help me ?

Reply:
solved

Reply:
Sorry Manar, can I still help you ?
You can reach me on Slack. I think it is easier for students to connect with Code Reviewers.